### PR TITLE
Clean up bcc_exception.h

### PR DESCRIPTION
**Pull request by palmtenor**
- Remove unused headers and Macros.
- Implement a `StatusTuple` class that has `code()` and `msg()` method, instead of using `std::get` and `std::tuple`, as suggested by @4ast and @drarmstr in #781.
- `clang-format -i` on the file.

### CODE REVIEW

**Comment by palmtenor** [comment]
Updated according to comment

File: src/cc/bcc_exception.h [hunk 1]
```diff
@@ -39,30 +36,28 @@ static inline StatusTuple mkstatus(int ret, const char *msg) {
...
 }
 
-#define TRY(CMD) \
-  do { \
-    int __status = (CMD); \
-    if (__status != 0) { \
-      return __status; \
-    } \
-  } while (0)
+static int status_code(const StatusTuple &status_tuple) {
```
Comment (4ast) [inline src/cc/bcc_exception.h]: since it's in the header file, it probably needs 'static inline' ?

**Comment by palmtenor** [comment]
Should I rename `mkstatus_` to be `mkstatus`? I guess originally the underscore is to avoid name conflict with the `mkstatus` functions from `exception.h`. But we don't have those anymore...

**Comment by 4ast** [comment]
you could have kept mkstatus() as inline func that returns StatusTuple, but since you've changed all of the callsites anyway, I think it's fine.

**Pull request merged**